logcat: ISensorManager Tried enabling a sensor

Logcat is spammed every second with:

```
12-23 15:24:03.790  1348 12976 E SensorService: android.frameworks.sensorservice.ISensorManager Tried enabling a sensor (LSM6DSV Accelerometer-Uncalibrated) without holding android.permission.OTHER_SENSORS
12-23 15:24:03.798  1348 12974 E SensorService: android.frameworks.sensorservice.ISensorManager Tried enabling a sensor (LSM6DSV Gyroscope-Uncalibrated) without holding android.permission.OTHER_SENSORS
12-23 15:24:03.800  1348 12976 E SensorService: android.frameworks.sensorservice.ISensorManager Tried enabling a sensor (ICP20100 Pressure Sensor) without holding android.permission.OTHER_SENSORS
12-23 15:24:04.856  1348 12976 E JavaBinder: *** Uncaught remote exception! Exceptions are not yet supported across processes. Client PID 1112 UID 1021.
12-23 15:24:04.856  1348 12976 E JavaBinder: java.lang.RuntimeException: android.os.RemoteException: Couldn't get targetSdkVersion for package android.frameworks.sensorservice.ISensorManager
12-23 15:24:04.856  1348 12976 E JavaBinder:    at android.os.Parcel.writeException(Parcel.java:2844)
12-23 15:24:04.856  1348 12976 E JavaBinder:    at android.os.Binder.execTransactInternal(Binder.java:1385)
12-23 15:24:04.856  1348 12976 E JavaBinder:    at android.os.Binder.execTransact(Binder.java:1302)
12-23 15:24:04.856  1348 12976 E JavaBinder: Caused by: android.os.RemoteException: Couldn't get targetSdkVersion for package android.frameworks.sensorservice.ISensorManager
12-23 15:24:04.856  1348 12976 E JavaBinder:    at com.android.server.pm.PackageManagerNative.getTargetSdkVersionForPackage(PackageManagerNative.java:111)
12-23 15:24:04.856  1348 12976 E JavaBinder:    at android.content.pm.IPackageManagerNative$Stub.onTransact(IPackageManagerNative.java:247)
12-23 15:24:04.856  1348 12976 E JavaBinder:    at android.os.Binder.execTransactInternal(Binder.java:1366)
12-23 15:24:04.856  1348 12976 E JavaBinder:    ... 1 more

```

Checking `Settings > Privacy > Permission Manager > Sensors > Show system`: all apps allowed

----

Build: UQ1A.231205.015.2023121200
Device: Google Pixel 7 Pro

--- a/services/sensorservice/SensorService.cpp
+++ b/services/sensorservice/SensorService.cpp
@@ -2210,6 +2210,16 @@ bool SensorService::canAccessSensor(const Sensor& sensor, const char* operation,
             canAccess = true;
         }
     } else {
+        if (sensor.getRequiredPermission() == "android.permission.OTHER_SENSORS") {
+            if (IPCThreadState::self()->getCallingUid() < AID_APP_START) {
+                // System processes do not expect that sensors that are protected by OTHER_SENSORS
+                // on GrapheneOS require a permission.
+                //
+                // The lack of this check led to crashes of the closed-source gpsd daemon.
+                return true;
+            }
+        }
+
         int targetSdkVersion = getTargetSdkVersion(opPackageName);
         if (targetSdkVersion > 0 && targetSdkVersion <= __ANDROID_API_P__ &&
             (sensor.getType() == SENSOR_TYPE_STEP_COUNTER ||

